assert: provide info about actual error

BridgeAR opened the pull request:
In case a error is caught in `assert.doesNotThrow` or
`assert.doesNotReject` it will now also indicate what the real
error message was.

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

lpinca commented on test/parallel/test-assert.js:
```diff
@@ -839,7 +832,7 @@ common.expectsError(
       type: assert.AssertionError,
       code: 'ERR_ASSERTION',
-      message: 'Got unwanted exception.\nundefined'
+      message: 'Got unwanted exception.\nActual message: undefined'
```
This still uses `common.expectsError()`, is this wanted?

ryzokuken replied:
Exactly. This call seems similar to https://github.com/nodejs/node/pull/19884/files#diff-81c8c31c5728ec9ed47700e55d9ceebfL127.

Can't this be changed to `assert.throws(...)` as well?

BridgeAR commented:
CI https://ci.nodejs.org/job/node-test-pull-request/14155/

BridgeAR commented:
Fixed failing test and I added quotes around the actual message to highlight what is part of it. New CI https://ci.nodejs.org/job/node-test-pull-request/14164/

BridgeAR commented:
Landed in 2bee7996a57a44aec1f7ee3a07cd49a5b235ad78
